src/appReview/slice.ts

```diff
@@ -0,0 +1,42 @@
+import { createSlice } from '@reduxjs/toolkit'
+import InAppReview from 'react-native-in-app-review'
+import { REHYDRATE, RehydrateAction } from 'redux-persist'
+import { getRehydratePayload } from 'src/redux/persist-helper'
+
+export interface State {
+  inAppRatingSupported: boolean
+  appRated: boolean
+  lastInteractionTimestamp: number | null
+}
+
+const initialState: State = {
+  // Android version >= 21 and iOS >= 10.3
+  // https://github.com/MinaSamir11/react-native-in-app-review#usage
+  inAppRatingSupported: !!InAppReview.isAvailable,
+  appRated: false,
+  lastInteractionTimestamp: null,
+}
+
+export const slice = createSlice({
+  name: 'appReview',
+  initialState,
+  reducers: {
+    inAppRatingSupported: (state, action) => {
+      state.inAppRatingSupported = action.payload
+    },
+    setAppRated: (state, action) => {
```

Comment:
is this used somewhere?

Reply:
This isn't use anywhere yet but before we dispatch a app rating action we'll want to check that in app rating is supported on the device via selector.